api/rest/restapi.go

```diff
@@ -666,6 +678,41 @@ func (api *API) unpinHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (api *API) pinPathHandler(w http.ResponseWriter, r *http.Request) {
+	var ci types.CidSerial
+	pinPath, ps := api.parsePath(w, r)
+	logger.Debugf("rest api pinPathHandler: %s", pinPath)
+
+	err := api.rpcClient.Call(
+		"",
+		"Cluster",
+		"PinPath",
+		types.PinSerialWithPath{
+			PinOpts: ps,
+			Path:    pinPath,
+		},
+		&ci,
+	)
+
+	api.sendResponse(w, http.StatusOK, err, &ci)
+	logger.Debug("rest api pinPathHandler done")
+}
+
+func (api *API) unpinPathHandler(w http.ResponseWriter, r *http.Request) {
+	var ci types.CidSerial
+	unpinPath, _ := api.parsePath(w, r)
+	logger.Debugf("rest api unpinPathHandler: %s", unpinPath)
+	err := api.rpcClient.Call(
+		"",
+		"Cluster",
+		"UnpinPath",
+		unpinPath,
+		&ci,
```

Comment:
Should also return a PinSerial, corresponding to the item we deleted.

Reply:
I have made the change the change to return PinSerial, but tbh I don't I a point in doing that. Why would user be interested in anything accept the cid that got unpinned?

Reply:
The user should know what actual Cid got unpinned since it could resolve to different things in different moments (/ipns).